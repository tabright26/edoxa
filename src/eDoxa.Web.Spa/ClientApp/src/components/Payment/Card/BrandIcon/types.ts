import { SizeProp } from "@fortawesome/fontawesome-svg-core";

export type BrandProp = "visa" | "amex" | "mastercard" | "discover";

export interface CardBrandIconProps {
  className: string;
  brand: BrandProp;
  size?: SizeProp;
}
