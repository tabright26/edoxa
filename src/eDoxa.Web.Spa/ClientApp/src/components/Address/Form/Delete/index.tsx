import React, { FunctionComponent } from "react";
import { Label, FormGroup, Form } from "reactstrap";
import { reduxForm, InjectedFormProps } from "redux-form";
import Button from "components/Shared/Button";
import { DELETE_USER_ADDRESS_FORM } from "utils/form/constants";
import { compose } from "recompose";
import { ValidationSummary } from "components/Shared/ValidationSummary";
import { deleteUserAddress } from "store/actions/identity";
import { throwSubmissionError } from "utils/form/types";
import { AxiosActionCreatorMeta } from "utils/axios/types";
import { AddressId } from "types";

interface FormData {}

interface OutterProps {
  addressId: AddressId;
  handleCancel: () => void;
}

type InnerProps = InjectedFormProps<FormData, Props>;

type Props = InnerProps & OutterProps;

const CustomForm: FunctionComponent<Props> = ({
  handleSubmit,
  error,
  handleCancel,
  submitting,
  anyTouched
}) => (
  <Form onSubmit={handleSubmit}>
    <ValidationSummary anyTouched={anyTouched} error={error} />
    <Label className="text-muted">Do you want to remove your address?</Label>
    <FormGroup className="mb-0">
      <Button.Submit loading={submitting} size="sm" className="mr-2">
        Remove
      </Button.Submit>
      <Button.Cancel onClick={handleCancel} />
    </FormGroup>
  </Form>
);

const enhance = compose<InnerProps, OutterProps>(
  reduxForm<FormData, Props>({
    form: DELETE_USER_ADDRESS_FORM,
    onSubmit: async (_values, dispatch, { addressId }) => {
      try {
        return await new Promise((resolve, reject) => {
          const meta: AxiosActionCreatorMeta = { resolve, reject };
          dispatch(deleteUserAddress(addressId, meta));
        });
      } catch (error) {
        throwSubmissionError(error);
      }
    },
    onSubmitSuccess: (result, dispatch, { handleCancel }) => handleCancel()
  })
);

export default enhance(CustomForm);
