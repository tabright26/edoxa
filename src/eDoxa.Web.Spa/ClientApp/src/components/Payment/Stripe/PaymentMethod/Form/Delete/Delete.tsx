import React, { FunctionComponent } from "react";
import { Label, FormGroup, Form } from "reactstrap";
import { reduxForm, InjectedFormProps } from "redux-form";
import Button from "components/Shared/Button";
import { DELETE_STRIPE_PAYMENTMETHOD_FORM } from "forms";
import { compose } from "recompose";
import FormValidation from "components/Shared/Form/Validation";
import { detachStripePaymentMethod } from "store/actions/payment";
import {
  StripePaymentMethodsActions,
  DETACH_STRIPE_PAYMENTMETHOD_FAIL
} from "store/actions/payment/types";
import { throwSubmissionError } from "utils/form/types";

interface FormData {}

interface OutterProps {
  paymentMethodId: string;
  handleCancel: () => void;
}

type InnerProps = InjectedFormProps<FormData, Props>;

type Props = InnerProps & OutterProps;

const ReduxForm: FunctionComponent<Props> = ({
  handleSubmit,
  error,
  handleCancel
}) => (
  <Form onSubmit={handleSubmit}>
    {error && <FormValidation error={error} />}
    <Label className="mb-3">
      Are you sure you want to delete this payment method?
    </Label>
    <FormGroup className="mb-0">
      <Button.Yes type="submit" className="mr-2" />
      <Button.No onClick={() => handleCancel()} />
    </FormGroup>
  </Form>
);

const enhance = compose<InnerProps, OutterProps>(
  reduxForm<FormData, Props>({
    form: DELETE_STRIPE_PAYMENTMETHOD_FORM,
    onSubmit: async (_values, dispatch: any, { paymentMethodId }) =>
      await dispatch(detachStripePaymentMethod(paymentMethodId)).then(
        (action: StripePaymentMethodsActions) => {
          switch (action.type) {
            case DETACH_STRIPE_PAYMENTMETHOD_FAIL: {
              throwSubmissionError(action.error);
              break;
            }
          }
        }
      ),
    onSubmitSuccess: (result, dispatch, { handleCancel }) => handleCancel()
  })
);

export default enhance(ReduxForm);
