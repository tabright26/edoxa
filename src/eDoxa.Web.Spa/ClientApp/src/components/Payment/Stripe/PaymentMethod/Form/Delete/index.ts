import DeleteStripePaymentMethodContainer from "./Container";

export default DeleteStripePaymentMethodContainer;
