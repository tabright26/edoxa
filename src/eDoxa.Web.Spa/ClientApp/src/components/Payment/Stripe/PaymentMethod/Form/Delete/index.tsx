import React, { FunctionComponent } from "react";
import { Label, FormGroup, Form, Button } from "reactstrap";
import { reduxForm, InjectedFormProps } from "redux-form";
import { DELETE_STRIPE_PAYMENTMETHOD_FORM } from "utils/form/constants";
import { compose } from "recompose";
import { ValidationSummary } from "components/Shared/ValidationSummary";
import { detachStripePaymentMethod } from "store/actions/payment";
import { DETACH_STRIPE_PAYMENTMETHOD_FAIL } from "store/actions/payment/types";
import { throwSubmissionError } from "utils/form/types";
import { RootActions } from "store/types";

interface FormData {}

interface OutterProps {
  paymentMethodId: string;
  handleCancel: () => void;
}

type InnerProps = InjectedFormProps<FormData, Props>;

type Props = InnerProps & OutterProps;

const Delete: FunctionComponent<Props> = ({
  handleSubmit,
  error,
  handleCancel
}) => (
  <Form onSubmit={handleSubmit}>
    <ValidationSummary error={error} />
    <Label className="mb-3">
      Are you sure you want to delete this payment method?
    </Label>
    <FormGroup className="mb-0">
      <Button type="submit" size="sm" color="primary" className="mr-2">
        Yes
      </Button>
      <Button onClick={() => handleCancel()} size="sm">
        No
      </Button>
    </FormGroup>
  </Form>
);

const enhance = compose<InnerProps, OutterProps>(
  reduxForm<FormData, Props>({
    form: DELETE_STRIPE_PAYMENTMETHOD_FORM,
    onSubmit: async (_values, dispatch: any, { paymentMethodId }) =>
      await dispatch(detachStripePaymentMethod(paymentMethodId)).then(
        (action: RootActions) => {
          switch (action.type) {
            case DETACH_STRIPE_PAYMENTMETHOD_FAIL: {
              throwSubmissionError(action.error);
              break;
            }
          }
        }
      ),
    onSubmitSuccess: (result, dispatch, { handleCancel }) => handleCancel()
  })
);

export default enhance(Delete);
