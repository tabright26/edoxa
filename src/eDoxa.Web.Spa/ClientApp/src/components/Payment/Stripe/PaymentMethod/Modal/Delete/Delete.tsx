import React, { FunctionComponent } from "react";
import { Modal, ModalBody, ModalHeader } from "reactstrap";
import { connectModal, InjectedProps } from "redux-modal";
import StripePaymentMethodForm from "components/Payment/Stripe/PaymentMethod/Form";
import { DELETE_STRIPE_PAYMENTMETHOD_MODAL } from "utils/modal/constants";
import { compose } from "recompose";
import { StripePaymentMethod } from "types";

type InnerProps = InjectedProps & {
  paymentMethod: StripePaymentMethod;
};

type OutterProps = {};

type Props = InnerProps & OutterProps;

const CustomModal: FunctionComponent<Props> = ({
  show,
  handleHide,
  paymentMethod
}) => (
  <Modal
    unmountOnClose={false}
    backdrop="static"
    centered
    isOpen={show}
    toggle={handleHide}
  >
    <ModalHeader toggle={handleHide}>DELETE PAYMENT METHOD</ModalHeader>
    <ModalBody>
      <StripePaymentMethodForm.Delete
        paymentMethodId={paymentMethod.id}
        handleCancel={() => handleHide()}
      />
    </ModalBody>
  </Modal>
);

const enhance = compose<InnerProps, OutterProps>(
  connectModal({
    name: DELETE_STRIPE_PAYMENTMETHOD_MODAL,
    destroyOnHide: false
  })
);

export default enhance(CustomModal);
