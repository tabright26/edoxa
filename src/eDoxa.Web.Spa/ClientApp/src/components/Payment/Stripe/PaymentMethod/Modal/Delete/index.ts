import CustomModal from "./Delete";

export default CustomModal;
