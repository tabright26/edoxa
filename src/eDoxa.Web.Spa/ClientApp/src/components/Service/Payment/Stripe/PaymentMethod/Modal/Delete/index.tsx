import React, { FunctionComponent } from "react";
import { Modal, ModalBody, ModalHeader } from "reactstrap";
import { connectModal, InjectedProps } from "redux-modal";
import StripePaymentMethodForm from "components/Service/Payment/Stripe/PaymentMethod/Form";
import { compose } from "recompose";
import { DELETE_STRIPE_PAYMENTMETHOD_MODAL } from "utils/modal/constants";
import { StripePaymentMethod } from "types/payment";

type InnerProps = InjectedProps & {
  paymentMethod: StripePaymentMethod;
};

type OutterProps = {};

type Props = InnerProps & OutterProps;

const Delete: FunctionComponent<Props> = ({
  show,
  handleHide,
  paymentMethod
}) => (
  <Modal backdrop="static" centered isOpen={show} toggle={handleHide}>
    <ModalHeader
      className="text-uppercase my-auto bg-gray-900"
      toggle={handleHide}
    >
      DELETE CREDIT CARD
    </ModalHeader>
    <ModalBody>
      {show && (
        <StripePaymentMethodForm.Delete
          paymentMethodId={paymentMethod.id}
          handleCancel={() => handleHide()}
        />
      )}
    </ModalBody>
  </Modal>
);

const enhance = compose<InnerProps, OutterProps>(
  connectModal({
    name: DELETE_STRIPE_PAYMENTMETHOD_MODAL,
    destroyOnHide: false
  })
);

export default enhance(Delete);
