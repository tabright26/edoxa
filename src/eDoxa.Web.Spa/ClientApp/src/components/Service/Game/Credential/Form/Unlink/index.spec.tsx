import React from "react";
import Delete from ".";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { GAME_LEAGUEOFLEGENDS } from "types/games";
import store from "store";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Delete game={GAME_LEAGUEOFLEGENDS} handleCancel={() => {}} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
