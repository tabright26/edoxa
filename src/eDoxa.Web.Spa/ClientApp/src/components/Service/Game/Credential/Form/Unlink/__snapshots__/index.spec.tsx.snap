// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="modal-body"
  >
    You can unlink your League of Legends credentials one month after linking your account.
  </div>
  <div
    className="modal-footer"
  >
    <button
      aria-label={null}
      className="mr-2 btn btn-primary btn-sm"
      onClick={[Function]}
      type="submit"
    >
      Yes
    </button>
    <button
      aria-label={null}
      className="btn btn-secondary btn-sm"
      onClick={[Function]}
      type="button"
    >
      No
    </button>
  </div>
</form>
`;
